### PR TITLE
[PSL-108] Process Open API request - `dupe detection` - in WN and SN

### PR DESCRIPTION
**Pull request by bacnh8**
Ref: 
- Sense flow https://pastel.wiki/en/Architecture/OpenAPI/Sense
- TicketStructure https://pastel.wiki/en/Architecture/Components/PastelOpenAPITicketStructures
- API Ticket: https://pastel.wiki/en/Architecture/Components/PastelOpenAPITicketStructures#sense-api-ticket

### CODE REVIEW

File: pastel/common.go [hunk 1]
```diff
@@ -114,3 +114,9 @@ type GetRawTransactionVerbose1Result struct {
 	Txid          string `json:"txid"`
 	Confirmations int64  `json:"confirmations"`
 }
+
+// GetActionFeesResult describes the result of "getactionfee getactionfees"
+type GetActionFeesResult struct {
+	SenseFee   float64 `json:"sensefee"`
+	CascadeFee float64 `json:"cascadefee"`
```
Comment (mateeullahmalik) [inline pastel/common.go:121]: please double check that json tag not snake case like sense_fee

File: supernode/services/senseregister/node.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	for i, node := range *nodes {
+		if node.ID == id {
+			*nodes = append((*nodes)[:i], (*nodes)[:i+1]...)
+			break
+		}
+	}
+}
+
+// Node represents a single supernode
+type Node struct {
```
Comment (mateeullahmalik) [inline supernode/services/senseregister/node.go:43]: I think we can combine Node & Nodes used al across project.

File: supernode/services/senseregister/service.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+	"github.com/pastelnetwork/gonode/pastel"
+	rqnode "github.com/pastelnetwork/gonode/raptorq/node"
+	"github.com/pastelnetwork/gonode/supernode/node"
+)
+
+const (
+	logPrefix = "sense"
+)
+
+// Service represent artwork service.
```
Comment (mateeullahmalik) [inline supernode/services/senseregister/service.go:25]: please correct the comment here. 
its sense not artwork

File: supernode/services/senseregister/service.go [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+type Service struct {
+	*task.Worker
+	*artwork.Storage
```
Comment (mateeullahmalik) [inline supernode/services/senseregister/service.go:28]: move storage to a common location? so sense doesn't import "artwork"?

File: walletnode/api/design/sense.go [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+		Description("Streams the state of the registration process.")
+		Meta("swagger:summary", "Streams state by task ID")
+
+		Payload(func() {
+			Extend(RegisterTaskPayload)
+		})
+		StreamingResult(ArtworkRegisterTaskState)
+
+		HTTP(func() {
+			GET("/start/{taskId}/state")
```
Comment (mateeullahmalik) [inline walletnode/api/design/sense.go:101]: I think "start" is not needed here?

**Review by mateeullahmalik — CHANGES_REQUESTED** [review]
Move some commonly used code (I've commented) out of artwork package to a common location to make it generic. Few minor things. 
Integration tests are failing as well.. Please fix.
Otherwise LGTM!
Nice work!

**Review by a-ok123 — DISMISSED** [review]
Conditionally approving, requires re-factoring and verifiaction

**Review by mateeullahmalik — APPROVED** [review]

**Review by a-ok123 — APPROVED** [review]

**Pull request merged**